packages/react-form/CHANGELOG.md

```diff
@@ -7,6 +7,10 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 <!-- ## [Unreleased] -->
 
+### Fixed
+
+- Fixed a TypeScript bug in FormMapping for FieldDictionary arrays [#1795](https://github.com/Shopify/quilt/pull/1795)
```

Comment:
Would this be a good place to tell people that `Promise<SubmitResult>` is still required when encountering this issue (if I'm following correctly?)?

Reply:
Explicitly adding `Promise<SubmitResult>` is only necessary if you define the submit handler outside of the config object. I've just realized that, because the `SubmitHandler` needs to return an object containing a string literal (`'success'` or ``'fail'`), it should really always be defined within the config object, and that is how the documentation examples work, too.

Sorry for including that red herring in my PR description 😅